### PR TITLE
Added Map.ChangeUnitColor()

### PR DESCRIPTION
**Pull request by Jesus-QC**
Credits to sanyae

### CODE REVIEW

**Comment by iRebbok** [comment]
Looks a little strange. What if a dev wants to change the color twice? Does it even have to be in the API?

**Comment by Jesus-QC** [comment]
Well.. Amathor asked me to put this in here but you are right, I will have to check if the unit has already changed the color

**Comment by Jesus-QC** [comment]
greeny green

**Comment by iRebbok** [comment]
@galaxy119 Your move

**Comment by galaxy119** [comment]
Looks solid to me, this has been possible in the game for several versions (since MP1 iirc). I'm fairly confident this won't be something removed from the game.

File: Exiled.API/Features/Map.cs [hunk 1]
```diff
@@ -514,6 +515,34 @@ public static DoorVariant GetDoorByName(string doorName)
...
+                {
+                    System.Text.RegularExpressions.Regex regex = new System.Text.RegularExpressions.Regex("<color=(.*)>(.*)</color>");
+                    var v = regex.Match(unit);
+                    string s = v.Groups[1].ToString();
+                    ply.ReferenceHub.characterClassManager.NetworkCurUnitName = unit.Replace(s, color);
+                }
+                else
+                {
+                    ply.ReferenceHub.characterClassManager.NetworkCurUnitName = $"<color={color}>{unit}</color>";
+                }
```
Comment (iRebbok) [inline Exiled.API/Features/Map.cs:542]: Simplify the regex to `<[^>]*?>` and call `Regex::Replace(string, string, string)` instead to remove tags (https://regex101.com/r/9Wsd6l/1)

**Review by iRebbok — CHANGES_REQUESTED** [review]

**Pull request merged**